docs/adr/0006-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,122 @@
+  - Both application-services team and the application teams generally put low priority on updating the application-services version.  We generally focus on the benefits of waiting to take releases rather than the costs.
+
+Mozilla generally favors a rapid release model to long cycle times.  Firefox desktop adopted rapid releases over a decade ago and there is current work to move both the mobile applications and our services towards rapid releases.  Although there are some costs to rapid releases, as a company we've decided that we're willing to accept them.
+
+Application services should consider moving towards a rapid release model.  The same reasoning that led other teams to adopt rapid releases also applies to us.  Furthermore, the mobile teams are moving towards rapid releases which may depend on us also adopting them.  Finally, we are in the process of moving towards a single monorepo for android code, with potential to add iOS code and/or move into the moz-central monorepo. This would require that application services code is taken by all other consumer applications in the monorepo as soon as it's merged.  Reducing the amount of time between our releases and when applications take those releases brings us closer to that monorepo world.
+
+## Decision Drivers
+
+* We want to get changes merged and released to our consumer applications faster.
+* We want to simplify pushing new code to applications, especially urgent bug fixes.
+* We want to move towards the future monorepo world.
+* Mozilla is moving towards rapid releases for mobile code.
+* We are willing to accept the overhead of upgrading application-services more frequently in application repositories.
+
+## Considered Options
+
+* **(A) Keep our current release process**
+* **(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+    * Phase 1: update our branching model and release process (Jan 2023 - Feb 2023)
+        * Update our branches to match how firefox-android works: a `main` branch that is used for the current nightly builds and a `releases_v[N]` for each previous release.
+        * Automate a nightly release of our `main` branch
+          * See the discussion at the bottom of this document for how this could work
+          * Note: regardless of which option we choose, we will no longer be using semver-style versions and also not following semver semantics.  We will be releasing breaking changes in a nightly cycle without bumping the major version.
```

Comment:
It would be great to call out that A-S would then follow the same versioning as all of M-C and mobile, which is a big win, in my opinion! We will no longer need to ask "but, which version of A-S is Fenix 110 using?"

Reply:
Yeah it'll be nicer for sure.  I added a pro in the list for this.

Note: I don't think we'll be able to keep the minor versions in sync between a-s and Fenix, since Fenix may want to release a minor version without any app-services changes.  Also, I think Fenix, iOS, and Desktop don't keep their minor versions in sync, which also would prevent this.  So I guess there will still be some digging needed for previous minor version releases, but it should be much simpler than the current situation.

Reply:
Thinking about this more (especially with a longer-term view of A-S potentially living in m-c some day with a separate-but-related version to the regular version.txt), we should probably have a test somewhere which enforces that the A-S major version matches the Gecko one.

Reply:
Hi, what would be the distinguishing factor for a release made for Firefox Android vs iOS when we generate dot releases?

Is the idea here that we put out a dot release that will basically work for all platforms but might have improvements/bugfixes for the specific platform it is intended for?

Reply:
Yes the idea is that when we did a dot release it would be for all platforms at once (Desktop too).

There's probably some scenarios when a particular app wouldn't take the new dot release.  For example, it was for a component that's not used.  But I think this should be pretty rare.  Even in the case where it's not going to affect the app, if the app was going to make a dot release anyways then I think they might as well take the latest app-services release.

Reply:
I think there's a distinction to be made between an app "taking" a new dot release as in "the PR was made, auto-merged, is effectively a no-op and will ship whenever it happens to" and "the PR was merged and we shipped a new release off of it".